infrastructure/ssz/src/main/java/tech/pegasys/teku/infrastructure/ssz/schema/impl/AbstractSszListSchema.java

```diff
@@ -75,6 +79,43 @@ protected TreeNode createTree(TreeNode dataNode, int length) {
     return BranchNode.create(dataNode, toLengthNode(length));
   }
 
+  @Override
+  public SszListT nCopies(final ElementDataT value, final int length) {
+    final int elementsPerChunk = getElementsPerChunk();
+    final TreeNode fullLeafNode;
+    TreeNode partialNode = null;
+    if (elementsPerChunk == 1) {
+      fullLeafNode = value.getBackingNode();
+    } else {
+      final Bytes singleItemData = ((LeafDataNode) value.getBackingNode()).getData();
+      final int partialNodeItemCount = length % elementsPerChunk;
+      if (singleItemData.isZero()) {
+        fullLeafNode = LeafNode.ZERO_LEAVES[singleItemData.size() * elementsPerChunk];
+        if (partialNodeItemCount != 0) {
+          partialNode = LeafNode.ZERO_LEAVES[singleItemData.size() * partialNodeItemCount];
+        }
+      } else {
+        fullLeafNode =
+            LeafNode.create(
+                Bytes.concatenate(Collections.nCopies(elementsPerChunk, singleItemData)));
+        if (partialNodeItemCount != 0) {
+          partialNode =
+              LeafNode.create(
+                  Bytes.concatenate(Collections.nCopies(partialNodeItemCount, singleItemData)));
+        }
+      }
+    }
+
+    List<TreeNode> leafNodes = Collections.nCopies(length / elementsPerChunk, fullLeafNode);
+    if (partialNode != null) {
+      leafNodes = new ArrayList<>(leafNodes);
+      leafNodes.add(partialNode);
+    }
```

Comment:
After applying #5350 does it make sense to handle the logic separately in `SszPrimitiveListSchemaImpl`and `SszListSchemaImpl`?